crates/eframe/src/native/run.rs

```diff
@@ -350,9 +362,16 @@ fn run_and_exit(event_loop: EventLoop<UserEvent>, mut winit_app: impl WinitApp +
                         .map(|window| window.request_redraw())
                 });
 
-            control_flow.set_wait_until(next_repaint_time);
+            event_loop_window_target.set_control_flow(ControlFlow::WaitUntil(next_repaint_time));
         };
-    })
+    });
+
+    std::process::exit(if let Err(e) = result {
+        log::warn!("Error from event loop: {e}");
+        1
+    } else {
+        0
+    });
```

Comment:
This doesn't exit cleanly on Android, perhaps not expose `run_and_exit()` on Android (not strictly related to the winit 0.29 upgrade).

Reply:
I think you're supposed return out of `fn main()` and let `android-activity` perform the cleanup:

https://github.com/rust-mobile/android-activity/blob/9fce89021959a6f6ea8853221367bfa305803369/android-activity/src/native_activity/glue.rs#L891-L901

Reply:
Right, so that's why perhaps `run_and_exit()` should perhaps not be available on Android, in favour of using `run_and_return()` there.

Reply:
https://docs.rs/winit/latest/winit/platform/run_on_demand/trait.EventLoopExtRunOnDemand.html :)

Reply:
Fixed - the function now returns an error